tests/parquet_test.py

```diff
@@ -16,20 +17,26 @@ def test_parquet(self):
         # get the dset from the dictionary in multi-locale cases
         for f in glob.glob('pq_test*'):
             os.remove(f)
-            self.assertTrue((ak_arr ==  pq_arr).all())
+            a = ak_arr.to_ndarray().sort()
```

Comment:
In the Chapel code, there is something like this going on in the HDF5 (and Parquet) code:
```
var tmp = glob(filelist[0]);
sort(tmp);
```
but it seems that in the `io_test.py`, they are doing the same `.to_ndarray()` and `.sort()` calls as I had added in here before comparing against the original arrays. Doing a quick test, it seems that removing those sorts does not cause the `io_test` to fail, so maybe it would make sense to remove those for the HDF5 test, but there is still then the issue with the Parquet that I will have to explore a bit more. That being said, the `io_test` is admittedly a bit difficult for me to understand, so I could be missing something there.

Reply:
This ended up being a rather silly issue, I had been appending just the locale number to the end of the filename, rather than the `%04i` that HDF5 is doing, so it was sorting `filename_LOCALE2` after `filename_LOCALE10`, since it is string sorting... 

Should be fixed now!

Reply:
Who invented numbers anyway...?

Reply:
> Who invented numbers anyway...?

And where do we file bug reports? :)